File: cpp_euler/p002.cpp
/* Project Euler - Problem 2
  By considering the terms in the Fibonacci sequence whose values do
  not exceed four million, find the sum of the even-valued terms.
  
  p002.cpp
  g++ p002.cpp -o p2.exe
*/
#include <iostream>

int main() {
  int i = 1, j = 2, e;
  int sum = 2;

  while (sum < 4E6){
    e = i+j;
    if (e % 2 == 0)
      sum += e;
    // get the values for the next term of the fib. seq.
    i=j;
    j=e;
  }

  std::cout << sum << std::endl; // 4613732

  return 0;
}

File: java_euler/p002.java
/**
 * Project Euler - Problem 1
 * Consider the terms in the Fibonacci sequence whose values do not exceed four million.
 * Find the sum of the even-valued terms.
 * 
 * p002.java
 * javac p002.java
 * java p002
 */

public class p002 {
    public static void main(String[] args) {
	System.out.println(new p002().run());
    }

    public String run() {
	int sum = 0;
	int i = 1;
	int j = 2;
	while (i <= 4000000) {
	    if (i%2 == 0)
		sum += i;
	    int temp = i+j;
	    i = j;
	    j = temp;
	}
	return Integer.toString(sum);
    }
}
